packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -649,6 +649,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // This code is plain wrong. It lies that it never returns undefined!!!
         // All callers should be fixed, as this API is called in detached state of container when we have
         // no storage and it's passed down the stack without right typing.
+        // back-compat 0.40 NoContainerStorageAdapter
```

Comment:
What does this comment mean?

Reply:
Renamed it to "NoStorageInDetachedMode" which means due to back compat we have to still have this check and remove after 2 iterations probably.